google/cloud/bigquery/bigquery_rest.cmake

```diff
@@ -63,10 +65,34 @@ target_compile_options(google_cloud_cpp_bigquery_rest
 add_library(google-cloud-cpp::experimental-bigquery_rest ALIAS
             google_cloud_cpp_bigquery_rest)
 
+# Create a header-only library for the mocks.
+add_library(google_cloud_cpp_bigquery_rest_mocks INTERFACE)
+target_sources(
+    google_cloud_cpp_bigquery_rest_mocks
+    INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/v2/minimal/mocks/mock_job_connection.h
+)
+target_link_libraries(
+    google_cloud_cpp_bigquery_rest_mocks
+    INTERFACE google-cloud-cpp::experimental-bigquery_rest GTest::gmock_main
+              GTest::gmock GTest::gtest)
+set_target_properties(
+    google_cloud_cpp_bigquery_rest_mocks
+    PROPERTIES EXPORT_NAME google-cloud-cpp::bigquery_rest_mocks)
```

Comment:
nit: something that depends on an experimental library is experimental, too. We should add the `experimental-` prefix to this